packages/nimble-components/src/wafer-map/modules/computations.module.ts

```diff
@@ -0,0 +1,189 @@
+import { scaleBand, ScaleLinear, scaleLinear } from 'd3-scale';
+import type { WaferMapDie } from '../types';
+import { Dimensions, Margin, WaferMapQuadrant } from '../types';
+
+interface MapDimensions {
+    origin: {
+        x: number,
+        y: number
+    };
+    rows: number;
+    cols: number;
+}
+
+/**
+ * Computations module
+ */
+export class Computations {
+    public containerDimensions: Dimensions;
+    public dieDimensions: Dimensions = { width: 0, height: 0 };
+    public radius: number;
+    public margin: Margin;
+
+    public horizontalScale: ScaleLinear<number, number>;
+    public verticalScale: ScaleLinear<number, number>;
+
+    private readonly baseMargin: Margin = {
+        top: 20,
+        right: 20,
+        bottom: 20,
+        left: 20
+    };
+
+    private readonly dieSizeFactor = 1.5;
+    private readonly defaultAlign = 0.5;
+
+    public constructor(
+        dies: WaferMapDie[],
+        axisLocation: WaferMapQuadrant,
+        canvasDimensions: Dimensions
```

Comment:
It looks like we don't mutate any of the inputs coming in, just read the values. We can declare that using the [Readonly](https://www.typescriptlang.org/docs/handbook/utility-types.html#readonlytype) utility

```suggestion
        dies: Readonly<Readonly<WaferMapDie>[]>,
        axisLocation: Readonly<WaferMapQuadrant>,
        canvasDimensions: Readonly<Dimensions>
```